api/routes/units.js

```diff
@@ -0,0 +1,238 @@
+const router = require('express').Router({ mergeParams: true });
+const Units = require('../models/unit.js');
+
+/**
+ * For the statements below i tried to re-factor/consolidate my code into a series of if/else if statements that would check for a query param being present, and find documents with the param key/value present. 
+ * If no params in request, return all documents.
+ * 
+ * However, I think this would still require further re-factoring if there is a feasible scenario where more than one of these params could be submitted i.e. to filter specific results like floor=4 AND occupied = false
+ * 
+ * GET /api/v1/units
+ *     Return a list of all of the units with all related information.
+ * GET /api/v1/units?kind=[kind]
+ *      Return a list of all units where the kind is equal to the provided kind. If none are found, return an empty array.
+ * GET /api/v1/units?floor=[integer]
+ *     Return a list of all units that are on the provided floor. If none are found, return an empty array.
+ * GET /api/v1/units?occupied=[true/false]
+ *       If true, return a list of all units that have a company associated with them. 
+ *      If false, return a list of all units that have no company associated with them. 
+ *       If none are found in either case, return an empty array.
+ */
+
+router.get('/', async (req, res, next) => {
+    const status = 200;
+    const {floor, kind, occupied} = req.query; // declare floor, occupied, kind params from req.query
+    if (floor) { // if query param is floor
+        const response = await Units.find({"unit.floor": floor}); // find units where unit.floor == floor
+        res.json({ status, response });
+    } else if (kind) { // otherwise, if query param is kind
+        const response = await Units.find({"unit.kind": kind}); // find units where unit.kind == kind; 
+        /**
+         * Question - I am having trouble with this and other excercises where the param has a space  e.g. "small office". 
+         * I tried using %20, + in the query param as well as just a true space, but no matter what i could not get a match.
+         * What might i be doing wrong here?
```

Comment:
This looks pretty good to me! One thing to look for is to see if you have `""`s as part of the query response. As we saw earlier, this can cause some issues. Please attend an office hours or set up a time with me and we can try and figure this out!